Lib/_collections_abc.py

```diff
@@ -444,18 +444,41 @@ def __create_ga(cls, origin, args):
         return super().__new__(cls, origin, ga_args)
 
     def __repr__(self):
-        if len(self.__args__) == 2 and self.__args__[0] is Ellipsis:
+        if len(self.__args__) == 2 and (self.__args__[0] is Ellipsis
+                                        or _concat_or_paramspec(self.__args__[0])):
             return super().__repr__()
         return (f'collections.abc.Callable'
                 f'[[{", ".join([_type_repr(a) for a in self.__args__[:-1]])}], '
                 f'{_type_repr(self.__args__[-1])}]')
 
     def __reduce__(self):
         args = self.__args__
-        if not (len(args) == 2 and args[0] is Ellipsis):
+        if not (len(args) == 2 and (args[0] is Ellipsis
+                                    or _concat_or_paramspec(args[0]))):
             args = list(args[:-1]), args[-1]
         return _CallableGenericAlias, (Callable, args)
 
+    def __getitem__(self, item):
+        # To allow the following::
+        #    P = ParamSpec('P)
+        #    C1 = Callable[P, T]
+        #    C1[[int, str], str] == Callable[[int, str], str]
+        ga = super().__getitem__(item)
+        args = ga.__args__
+        if not isinstance(ga.__args__[0], tuple):
```

Comment:
Oh oops I just remembered one more reason why I had to do this: 
After we subclassed Callable in the other PR, I forgot to override the ``__getitem__``, so it automatically returned a plain `types.GenericAlias` rather than the special subclass. The problem with that is that the `repr` is now wrong, as it shows the underlying flattened version rather than the one the user writes.

Eg.
```python
>>> Callable[[dict, float], T][int]
collections.abc.Callable[dict, float, int]
```
We probably need to backport some part of this change to the 3.9 branch to fix that.

Reply:
Ouch, i just noticed something else, the repr for both Callables in `typing` and `collections.abc` don't show `types.GenericAlias` properly either:
```python
>>> typing.Callable[dict[float, str], int]
typing.Callable[[dict[float, str]], int]
>>> Callable[dict[float, str], int]
collections.abc.Callable[[dict[float, str]], int]
```
Looks like something else we need to backport to 3.9